test/functional/wallet_conflicts.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""Test conflicts tracking with multilple txn conflicting a conflicted tx."""
+
+from decimal import Decimal
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+        assert_equal,
+)
+
+class TxConflicts(BitcoinTestFramework):
+    def add_options(self, parser):
+        self.add_wallet_options(parser)
+
+    def set_test_params(self):
+        self.num_nodes = 3
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        self.log.info("Send tx from which to conflict outputs later")
+        txid_conflict_from_1 = self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), Decimal("10"))
+        txid_conflict_from_2 = self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), Decimal("10"))
+        self.generate(self.nodes[0], 1)
+        self.sync_blocks()
+
+        self.log.info("Disconnect nodes to broadcast conflicts on their respective chains")
+        self.disconnect_nodes(0, 1)
+        self.disconnect_nodes(2, 1)
+
+        self.log.info("Create a transaction with conflicted transactions spending outpoints A & B")
+        nA = next(tx_out["vout"] for tx_out in self.nodes[0].gettransaction(txid_conflict_from_1)["details"] if tx_out["amount"] == Decimal("10"))
+        nB = next(tx_out["vout"] for tx_out in self.nodes[0].gettransaction(txid_conflict_from_2)["details"] if tx_out["amount"] == Decimal("10"))
+        inputs_conflicted_tx = []
+        inputs_conflicted_tx.append({"txid": txid_conflict_from_1, "vout": nA})
+        inputs_conflicted_tx.append({"txid": txid_conflict_from_2, "vout": nB})
+        inputs_conflicting_tx_A = []
+        inputs_conflicting_tx_A.append({"txid": txid_conflict_from_1, "vout": nA})
+        inputs_conflicting_tx_B = []
+        inputs_conflicting_tx_B.append({"txid": txid_conflict_from_2, "vout": nB})
+
+        outputs_conflicted_tx = {}
+        outputs_conflicted_tx[self.nodes[0].getnewaddress()] = Decimal("19.99998")
+        outputs_conflicting_tx_A = {}
+        outputs_conflicting_tx_A[self.nodes[0].getnewaddress()] = Decimal("9.99998")
+        outputs_conflicting_tx_B = {}
+        outputs_conflicting_tx_B[self.nodes[0].getnewaddress()] = Decimal("9.99998")
+
+        conflicted = self.nodes[0].signrawtransactionwithwallet(self.nodes[0].createrawtransaction(inputs_conflicted_tx, outputs_conflicted_tx))
+        conflicting_tx_A = self.nodes[0].signrawtransactionwithwallet(self.nodes[0].createrawtransaction(inputs_conflicting_tx_A, outputs_conflicting_tx_A))
+        conflicting_tx_B = self.nodes[0].signrawtransactionwithwallet(self.nodes[0].createrawtransaction(inputs_conflicting_tx_B, outputs_conflicting_tx_B))
```

Comment:
Sorry to bikeshed, but I think the naming is a bit hard to follow ("conflicted" vs "conflicting" don't immediately make sense to me, especially since both sets of transactions get conflicted at some point during the test). Might I suggest names that indicate which input(s) they spend, which node broadcasts them, and what "generation" they are? For example

`tx_AB_0_parent` instead of `conflicted`
`tx_A_1` instead of `conflicting_tx_A`
`tx_B_1` instead of `conflicting_tx_B`

`C` instead of `nA` for the 19.99998BTC output, because `nA` is already the name of the 10BTC output from one of the funding transactions.
`tx_C_0_child` instead of `child_conflicted`

Reply:
I agree, the naming here was a bit confusing. It should be fixed now.